core/domain/wipeout_service.py

```diff
@@ -1417,3 +1418,76 @@ def _pseudonymize_models_transactional(
                 pseudonymized_id)
+
+
+def _pseudonymize_version_history_models(pending_deletion_request):
+    """Pseudonymizes the version history models for the user with user_id.
+
+    Args:
+        pending_deletion_request: PendingDeletionRequest. The pending
+            deletion request object to be saved in the datastore.
+    """
+    user_id = pending_deletion_request.user_id
+
+    version_history_model_class = exp_models.ExplorationVersionHistoryModel
+    version_history_models = version_history_model_class.query(
+        user_id == version_history_model_class.committer_ids
+    ).fetch()
+    version_history_ids = {model.id for model in version_history_models}
+
+    # Here, we are not generating separate pseudonymized_id for all version
+    # history models because there might be a large number of models having
+    # association to user_id which increases the chance of collisions during
+    # id generation. Also, we don't need different pseudonymized ids for
+    # pseudonymizing the same user_id in different models.
+    pseudonymized_id = user_models.PseudonymizedUserModel.get_new_id('')
+    for entity_id in version_history_ids:
+        pending_deletion_request.pseudonymizable_entity_mappings[
+            models.NAMES.exploration.value][entity_id] = pseudonymized_id
+    save_pending_deletion_requests([pending_deletion_request])
+
+    @transaction_services.run_in_transaction_wrapper
+    def _pseudonymize_models_transactional(version_history_models):
+        """Pseudonymize user ID fields in the models.
+
+        This function is run in a transaction, with the maximum number of
+        version_history_models being MAX_NUMBER_OF_OPS_IN_TRANSACTION.
+
+        Args:
+            version_history_models: list(BaseModel). Models whose user IDs
+                should be pseudonymized.
+        """
+        for model in version_history_models:
+            # Pseudonymize user id from state_version_history.
+            for state_name in model.state_version_history:
+                if model.state_version_history[state_name][
+                    'committer_id'] == user_id:
+                    model.state_version_history[state_name]['committer_id'] = (
+                        version_history_ids_to_pids[model.id])
+
+            # Pseudonymize user id from metadata_version_history.
+            if model.metadata_version_history['committer_id'] == user_id:
+                model.metadata_version_history['committer_id'] = (
+                    version_history_ids_to_pids[model.id])
+
+            # Pseudonymize user id from committer_ids.
+            for idx, committer_id in enumerate(model.committer_ids):
+                if committer_id == user_id:
+                    model.committer_ids[idx] = (
+                        version_history_ids_to_pids[model.id])
+
+        version_history_model_class.update_timestamps_multi(
+            version_history_models)
```

Comment:
Does updating timestamps of this model have other effects? Are the commit histories on the history tab ordered by timestamp?

Reply:
Basically point being do we lose the data of the actual commit time if we update timestamps?

Reply:
This model does not store the commit times, hence, updating timestamps will not have any affect on them.